src/planet-a/Active/SysPlanetA.cpp

```diff
@@ -174,30 +186,53 @@ void SysPlanetA::debug_create_chunk_collider(osp::active::ActiveEnt ent,
 void SysPlanetA::update_geometry()
 {
 
-    auto view = m_scene.get_registry().view<ACompPlanet>();
+    auto view = m_scene.get_registry().view<ACompPlanet, ACompTransform>();
 
     for (osp::active::ActiveEnt ent : view)
     {
         auto &planet = view.get<ACompPlanet>(ent);
+        auto &tf = view.get<ACompTransform>(ent);
 
-        if (!planet.m_planet.is_initialized())
+        if (planet.m_planet.get() == nullptr)
         {
+            planet.m_planet = std::make_shared<PlanetGeometryA>();
+            PlanetGeometryA &rPlanetGeo = *(planet.m_planet);
+
             // initialize planet if not done so yet
             std::cout << "Initializing planet\n";
-            planet.m_planet.initialize(planet.m_radius);
+            planet.m_icoTree = std::make_shared<IcoSphereTree>();
+
+            planet.m_icoTree->initialize(planet.m_radius);
+
+            rPlanetGeo.initialize(planet.m_icoTree, 4, 1u << 9, 1u << 13);
+
+            planet.m_icoTree->event_add(planet.m_planet);
```

Comment:
Does it make sense to provide a constructor for IcoSphereTree that can accept the planet to add, so that it can be initialized in one go?

Reply:
The IcoSphereTree is designed to allow being shared between multiple PlanetGeometryA instances, even between separate scenes. It's likely that the initialization of IcoSphereTree will be moved to an entirely different function some time in the future.

Reply:
ok